### ISSUE TITLE
Unit tests for `SpriteButtonComponent`

### ISSUE THREAD

**Issue by spydon** [flame-engine/flame#1967]
# What could be improved

Currently the `SpriteButtonComponent` doesn't have any tests, it would be good to improve the test coverage here.

# Why should this be improved

So that we don't accidentally break anything when doing changes.

# More information

https://github.com/flame-engine/flame/blob/main/packages/flame/lib/src/components/input/sprite_button_component.dart

(This directory doesn't exist yet, but should be created with this issue)
https://github.com/flame-engine/flame/tree/main/packages/flame/test/input

# Good to know

Make sure that you comment on the issue that you will be working on this before starting to program.

To set up the Flame environment you need to follow the instructions in our [contributing.md](https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md#environment-setup) doc, if you have any problems setting up your environment (or other issues of course) feel free to ask for help in the Flame channel on our [Discord server](https://discord.com/invite/pxrBmy4).

**Comment by adityathakurxd** [comment]
Hi, Can take up this issue?

**Comment by spydon** [comment]
> Hi, Can take up this issue?

Assigned you to the issue, make sure to join our Discord if you have any questions. :)

**Comment by ShwetaChauhan18** [comment]
Hi, I know it's already assigned but still Can I take this issue ??

**Comment by spydon** [comment]
> Hi, I know it's already assigned but still Can I take this issue ??

If @adityathakurxd is not planning to finish this you can take this, otherwise I can recommend [this issue](https://github.com/flame-engine/flame/issues/1998) if you want to write up some unit tests.

I also see in your commit that you have added `@mustCallSuper` where we have missed them, you can upload that as a separate PR. Don't import `material` though, import `package:meta/meta.dart` instead.

**Comment by ShwetaChauhan18** [comment]
@adityathakurxd : Hi, let me know if you are not planning to complete this issue. I have already implemented my code but I will create PR if you are not planning to complete as you have taken this first.

**Comment by adityathakurxd** [comment]
Hi @ShwetaChauhan18, sure please go ahead and make the pull request if you're already done with the code. I am still trying to understand how to write unit tests. It might take me some time to get hold of that and contribute here. Still somewhat new to writing tests.

**Comment by ShwetaChauhan18** [comment]
Thanks @adityathakurxd 

@spydon: Can you please assign this issue to me

**Comment by spydon** [comment]
> Thanks @adityathakurxd 
> 
> @spydon: Can you please assign this issue to me
> 

Done!
